In [5]:
import pandas as pd

# Muat Datasets
train_path = "D:/File-File Kompetisi/Satria Data/data_training.csv"
test_path = "D:/File-File Kompetisi/Satria Data/data_testing_no_label.csv"
submission_path = "D:/File-File Kompetisi/Satria Data/Sample_Submission.csv"

# Baca data training 
df_train = pd.read_csv(train_path)
df_test = pd.read_csv(test_path)
df_submission = pd.read_csv(submission_path)

# Display informasi umum tentang data training
df_train.info(), df_train.head()

# Muat ulang datasets dengan delimiter yang sesuai
df_train = pd.read_csv(train_path, delimiter=';')
df_test = pd.read_csv(test_path, delimiter=';')

# Display informasi umum tentang data training
df_train.info(), df_train.head()

# Cek distribusi kelas target
df_train["Kategori_Gizi"].value_counts(normalize=True)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 1 columns):
 #   Column                                                                                                                                                                                                                 Non-Null Count  Dtype 
---  ------                                                                                                                                                                                                                 --------------  ----- 
 0   ID_Siswa;Usia;Berat_Badan;Tinggi_Badan;IMT;Asupan_Kalori;Aktivitas_Fisik;Kategori_Gizi;Tingkat_Kesulitan;Jenis_Kelamin;Frekuensi_Makan;Konsumsi_Sayur_Buah;Durasi_Tidur (Menit);Fast_Food_Per_Minggu;Riwayat_Penyakit  8000 non-null   object
dtypes: object(1)
memory usage: 62.6+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 15 columns):
 #   Column      

Kategori_Gizi
Gizi Lebih     0.451875
Gizi Kurang    0.341500
Gizi Baik      0.206625
Name: proportion, dtype: float64

Kesimpulan 
Gizi Lebih : 45.18 %
Gizi Kurang : 34.15 %
Gizi Baik : 20.66 % (Paling Sedikit) 

ImportError: cannot import name 'LabelEncoder' from 'sklearn_preprocessing' (C:\Users\Lenovo\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\LocalCache\local-packages\Python313\site-packages\sklearn_preprocessing\__init__.py)